packages/create-svelte/template/jsconfig.json

```diff
@@ -1,5 +1,8 @@
 {
 	"compilerOptions": {
+		"baseUrl": ".",
+		"checkJs": true,
```

Comment:
I wouldn't turn that on. People just using JS might want the untyped nature of js / are used to it and then get confused on how to get rid of these errors or where they come from.

Reply:
Do you think it's still a reasonable default for `tsconfig.json`?

Reply:
Good question.. I'm not even sure about `allowJs` anymore 😄 maybe we should remove both js flags, at least that's how it is for sapper/Svelte TS Templates currently.

Reply:
`allowJS` defaults to `false` and if we leave it out then people using the TypeScript template will suddenly be forced to use full-TS (unless they re-add it themselves), so I'm pretty sure that needs to be there at least.

Reply:
Well wouldn't people turning on TS want the full TypeScript experience? If I write TypeScript I would be surprised if there are JS files lying around in my project.

Reply:
One of the key points in the `create-svelte` CLI was "you can always add TypeScript later" (even though the CLI doesn't do that yet), and so that's one case where you'll have JS files lying around while migrating to TS.

Although this is a fresh kit setup, they might also want a mixed environment in order to have TypeScript server routes but JS in their `.svelte` files. `<script>` tags also default to JS without the `lang="ts"` because we don't set it in `svelte.config.cjs` preprocessor options when we add TS, and I assumed this was deliberate so as not to force TS `<script>` tags out of the box. Similarly, we don't default `<style>` preprocessing to LESS/SCSS when the options are picked, we default to CSS.

Reply:
Okay apparently the JS `<script>` tag in a `.svelte` is still accepted even in an `"allowJs": false` environment. `checkJs` notably still affects the checking of `<script>` tag of `.svelte` files in VS Code, so it's still useful to leave on in the TS template.

However, this configuration not allowed by the compiler:
```jsonc
{
  "allowJs": false,
  "checkJs": true,
  // ...
}
```
```
TS5052: Option 'checkJs' cannot be specified without specifying option 'allowJs'.
```
Which means either:
- We "disable" both `allowJs` and `checkJs` but it's still available in `.svelte` files without the advanced typechecking
- We enable both `allowJs` and `checkJs` and have it typechecked

The latter seems preferable for TypeScript users, I think?

Reply:
The reason it does not matter for TS importing Svelte files is that TS does not understand Svelte files, that's why the ambient module declaration of `"*.svelte"` is needed in `globals.d.ts`. I was going to say turning both off sounds better since it's typescript's default but you are right that turning both on means type-checking JS-Svelte files which might be desirable. So yeah.. maybe turn on both after all.

Reply:
I think the main reason we need `allowJs` is because the files for `$app` and `$navigation` are copied into the user's project, so I expect it will be necessary to get the types for the SvelteKit library. I think it's a bit odd that we do that and force `allowJs` on our users instead of making the SvelteKit library be a standard library under `node_modules`